core/arch/arm/mm/tee_pager.c

```diff
@@ -274,6 +286,40 @@ static void tee_pager_hide_pages(void)
 	/* TODO only invalidate entries touched above */
 	core_tlb_maintenance(TLBINV_UNIFIEDTLB, 0);
 }
+
+/*
+ * Find mapped pmem, hide and move to pageble pmem.
+ * Return false if page was not mapped, and true if page was mapped.
+ */
+static bool release_pageable_zi(vaddr_t page_va)
```

Comment:
Since this function is only called from pageable code it will end up in the pageable area. This function must not cause any page faults since it will lead to race conditions with the pager.

Something has to be done to guarantee that this function is in a non-pageable area.